File: EvalListener.js
// Add a listener to report values found
EvalListener = function(outText) {
  ExprListener.ExprListener.call(this); // inherit default listener
  this.theStack = [];  // Execution stack
  this.map = {};       // Variable map
  this.result = ''; // Output 
  return this;
};

// inherit default listener
EvalListener.prototype = Object.create(ExprListener.ExprListener.prototype);
EvalListener.prototype.constructor = EvalListener;

// override default listener behavior
EvalListener.prototype.exitId = function(ctx) {
  //LOGGER.info( "Exiting Id" );
  var id = ctx.ID().getText();
  // LOGGER.info( "  Pushing value of " + id + " to stack" );
  if ( this.map.hasOwnProperty(id) ) {
    this.theStack.push( this.map[id] );
  } else {
    // Variable not declared, push NaN to stack.
    this.theStack.push( NaN );
  }
}

EvalListener.prototype.exitAssign =
function(ctx) {
  this.map[ctx.ID().getText()] = this.theStack.pop();
};

EvalListener.prototype.exitInt = function(ctx) {
  // LOGGER.info( "Exiting Int" );
  this.theStack.push( parseInt( ctx.INT().getText() ));
}

EvalListener.prototype.exitPrint = function( ctx) {
  //LOGGER.info( "Exiting Print" );
  if ( !this.theStack.length < 1 ) {
    this.result += this.theStack.pop() + "\n";
  }
}

EvalListener.prototype.exitClear = function(ctx) {
  // LOGGER.info("Exiting Clear");
  this.theStack = [];
  this.map = {};
}

EvalListener.prototype.exitAddSub = function(ctx) {
  //LOGGER.info( "Exiting AddSub" );
  var b = this.theStack.pop();
  var a = this.theStack.pop();
  var result;
  if (ctx.op.type == ExprParser.ExprParser.ADD ) {
    result = a+b;
  } else {
    result = a-b;
  }
  this.theStack.push( result );
}

EvalListener.prototype.exitMulDiv = function(ctx) { 
  // LOGGER.info( "Exiting MulDiv" );
  b = this.theStack.pop();
  a = this.theStack.pop();
  var result;
  if (ctx.op.type == ExprParser.ExprParser.MUL) {
    result = a*b;
  } else {
    result = a/b;
  }
  this.theStack.push( result );
}

exports.EvalListener = EvalListener;

File: EvalListener.py
from expr.ExprListener import ExprListener
from expr.ExprParser import ExprParser


# Add a listener to report values found
class EvalListener(ExprListener):
    def __init__(self):
        self.theStack = []  # Execution stack
        self.map = {}  # Variable map

    # override default listener behavior
    def exitId(self, ctx):
        # LOGGER.info( "Exiting Id" );
        var_id = ctx.ID().getText()
        # LOGGER.info( "  Pushing value of " + var_id + " to stack" );
        if var_id in self.map:
            self.theStack.append(self.map[var_id])

    def exitAssign(self, ctx):
        self.map[ctx.ID().getText()] = self.theStack.pop()

    def exitInt(self, ctx):
        # LOGGER.info( "Exiting Int" );
        self.theStack.append(int(ctx.INT().getText()))

    def exitPrint(self, ctx):
        # LOGGER.info( "Exiting Print" );
        if not (len(self.theStack) < 1):
            print self.theStack.pop()

    def exitClear(self, ctx):
        # LOGGER.info("Exiting Clear");
        self.theStack = []
        self.map = {}

    def exitAddSub(self, ctx):
        # LOGGER.info( "Exiting AddSub" );
        b = self.theStack.pop()
        a = self.theStack.pop()
        if ctx.op.type == ExprParser.ADD:
            result = a + b
        else:
            result = a - b
        self.theStack.append(result)

    def exitMulDiv(self, ctx):
        # LOGGER.info( "Exiting MulDiv" );
        b = self.theStack.pop()
        a = self.theStack.pop()
        if ctx.op.type == ExprParser.MUL:
            result = a * b
        else:
            result = a / b  # Integer divisiion?
        self.theStack.append(result)
